csp-mixin/alerts/gcp-alerts.yml

```diff
@@ -14,3 +14,129 @@ groups:
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Compute Engine'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'IO latency is too high.'
+          description: 'Check {{ $labels.instance_id }} VM for I/O bottlenecks and upgrade to SSD if necessary.'
+          dashboard_uid: 'f115fe73641347c43415535d77e2dc0f'
+
+      - alert: GcpCloudSQLHighCpu
+        expr: |
+          100 * avg by (job,project_id,instance,database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_cpu_utilization{job=~".+",project_id=~".+",instance=~".+", database_id=~".+"}) > 80
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'CPU utilization is too high.'
+          description: 'Check {{ $labels.database_id }} database for high CPU queries and optimize them, or scale up the instance if sustained high usage.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLMemoryUsage
+        expr: |
+          100 * avg by (job,project_id,instance,database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_memory_utilization{job=~".+",project_id=~".+",instance=~".+", database_id=~".+"}) > 85
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'Memory utilization is too high.'
+          description: 'Review high-memory queries or add more memory to the {{ $labels.database_id }} instance.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLDiskUsage
+        expr: |
+          100 * avg by (job,project_id,instance,database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_disk_utilization{job=~".+",project_id=~".+",instance=~".+", database_id=~".+"}) > 85
+        for: 5m
+        keep_firing_for: 10m
+        labels:
+          severity: critical
+          service: 'Cloud SQL'
+          namespace: cloud-provider-gcp
+        annotations:
+          summary: 'Disk utilization is too high.'
+          description: 'Delete or archive unused data, or increase disk size to the {{ $labels.database_id }} database.'
+          dashboard_uid: 'cc710d49022fdd69bed0e992891863e9'
+
+      - alert: GcpCloudSQLActiveConnections
+        expr: |
+          avg by (job,project_id,instance, database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_threads{thread_kind="THREADS_CONNECTED", job=~".+",project_id=~".+",instance=~".+", database_id=~".+"}) > 0.9 * avg by (job,project_id,instance, database_id) (stackdriver_cloudsql_database_cloudsql_googleapis_com_database_mysql_max_connections{job=~".+",project_id=~".+",instance=~".+", database_id=~".+"})
```

Comment:
Based on image below the database_mysql_threads seems to be a total not a percentage. I think you need to do something like: `100 * (database_mysql_threads / mysql_max_connections)` and then compare the result with 90 %.

<img width="1192" alt="Screenshot 2024-11-28 at 09 10 08" src="https://github.com/user-attachments/assets/8a776332-ff9f-4a31-950c-9d77cf7d8cab">

Reply:
So I want to compare **threads connected** to be a larger number than 90% of the **max number of connections**.
This means:
**threads connected** > 0.9 * **max number of connections**.